src/configuration.ts

```diff
@@ -277,6 +276,11 @@ export class Configuration {
         OverridesStatus.ApprovedAll &&
       this.allSettingsMatch()
     ) {
+      // Delete previous entries
+      existingKeys.forEach((key) => {
+        this.context.globalState.update(key, undefined);
```

Comment:
This is for my own understanding -- does this remove the key from the global state entirely, or keep the key and set its value to `undefined`?

Reply:
In regular JS objects, it would keep the key with `undefined` as its value, but the `globalState` has logic to actually remove the key completely.